Sage/carbon
# Menu fixes

fix #6934, fix #7000, fix #7010

### Proposed behaviour


Ensures that `MenuItem`s flex so that they all have the same height if any wrap their content to new
lines at smaller screen resolutions. Ensures no additional padding is set on `MenuItem` children of
`MenuFullscreen`.

![image](https://github.com/user-attachments/assets/983170e3-eb83-4c3f-9e84-f8e9cb468de9)


BREAKING CHANGE: `Menu` no longer supports `height`, `minHeight`, `maxHeight`, `size`,
`overflowY` and `display` props. `MenuItem` no longer supports `height`, `minHeight`,
`maxHeight`, `size`, `verticalAlign`, `overflow`, `overflowY`, `overflowX` and
`display` props.

Adds invariant to ensure that the component does not render when `children` are passed but `submenu`
is an empty string

### Current behaviour


MenuItems do not wrap/flex uniformly
When `submenu` is an empty string the `children` are still rendered but within the parent `MenuItem`

### Checklist




#### QA


### Additional context



### Testing instructions


* `menu-test--when-menu-items-wrap` story added for testing when items wrap
* Worth doing brush test of all other stories to ensure no regressions
* To test issue 7001 add the following code into any menu test story and confirm the invariant has fired:

```
<Menu menuType="black">
      <MenuItem href="#">Menu Item One</MenuItem>
      <MenuItem onClick={() => {}} submenu="">
        <MenuItem
          href="#"
          onClick={() => {
            alert("clicked");
          }}
          variant="alternate"
        >
          Link
        </MenuItem>
        <MenuItem href="#">Submenu Item Two</MenuItem>
      </MenuItem>
    </Menu>
```
